src/lib.rs

```diff
@@ -477,6 +497,40 @@ impl Criterion {
         self
     }
 
+    /// Benchmarks multiple functions
+    ///
+    /// All functions get the same input and are compared with the other implementations.
+    /// Works similar to `bench`, but with multiple functions.
+    ///
+    /// ``` ignore-test
+    /// fn bench_seq_fib(b: &mut Bencher, i: &u32) {
+    ///     b.iter(|| {
+    ///         seq_fib(test::black_box(i));
```

Comment:
> Do you oppose the usage of black_box in general or just in this case?

IMO, ideally, the _users_ should never need to use `black_box` in their benchmarks. It's cumbersome, they need import the `test` crate and add a feature gate. I'm OK with sprinkling `black_box` in all of criterion's internals because that's not user facing.

But if your experience says that LLVM optimizes away everything if `black_box` is not used, then we should that to the users. I see two ways to do that:

- We document everything without `black_box` and add a NOTE that says that LLVM is too smart and may optimize your code away and that you may want to use `black_box` to avoid that.
- Or we do the reverse: Document everything with `black_box` and mention that it may not be necessary in some cases.

Thoughts?